src/Analysis/Ast/Impl/Types/PythonFunctionType.cs

```diff
@@ -141,19 +139,8 @@ public override PythonMemberType MemberType
 
         internal ImmutableArray<string> Dependencies { get; private set; } = ImmutableArray<string>.Empty;
 
-        internal void AddOverload(IPythonFunctionOverload overload) {
-            if (_overload != null) {
-                _overloads = new List<IPythonFunctionOverload> { _overload, overload };
-                _overload = null;
-                return;
-            }
-
-            if (_overloads != null) {
-                _overloads.Add(overload);
-            } else {
-                _overload = overload;
-            }
-        }
+        internal void AddOverload(IPythonFunctionOverload overload) 
+            => _overloads = _overloads.Count > 0 ? _overloads.Add(overload) : ImmutableArray<IPythonFunctionOverload>.Create(overload);
```

Comment:
Is there any performance benefit to this rather than just calling `Add` unconditionally? I think it needs to allocate a new backing array in any case.

Reply:
Hmm, I guess it skips over all of the compare exchange and such.

Reply:
There is a memory benefit. By default, `ImmutableArray<T>` acts like a `List<T>` - creates an underlying array of 4 elements. Since it is a common case here that we have only one element (at least how I understand original optimization), we can start by creating an array with just one element.